File: JavaEquivalent/src/task/one/PrintUserRoles.java
package task.one;
import java.util.Optional;

public class PrintUserRoles {
	
	private static void printRoles(User user){
		Optional.of(user).map(User::getRoles).get().forEach(System.out::println);
	}

	public static void main(String args[]) {
		
		//all kinds of possibilities
		User user1 = new User();
		User user2 = null;
		
		printRoles(user1);
		printRoles(user2);
	}

}

File: MyScalaPlayground/src/task/one/PrintUserRoles.scala
package task.one;

/**
 * a fake repository, in real world it could return an empty Option
 * get this from:
 * http://docs.scala-lang.org/tutorials/scala-for-java-programmers.html
 */
object PrintUserRoles {

  def printRoles(user: User) {
    Option(user) map { _.getRoles } map { _.foreach { println } }
  }

  def main(args: Array[String]) {

    //below are all kinds of possibilities
    val user1 = new User()
    val user2 = null

    printRoles(user1)
    printRoles(user2)

  }

}